azurerm/resource_arm_key_vault_key.go

```diff
@@ -212,6 +265,9 @@ func resourceArmKeyVaultKeyUpdate(d *schema.ResourceData, meta interface{}) erro
 		return nil
 	}
 
+	if d.HasChange("key_size") || d.HasChange("key_type") || d.HasChange("curve") {
+		return resourceArmKeyVaultKeyCreate(d, meta)
```

Comment:
I think the motivation here and with the `ForceNew` removal was to get the same behavior as with key vault secrets: new versions of the key instead of deleting and recreating the key (https://github.com/terraform-providers/terraform-provider-azurerm/blob/d308df8651194068ea339b8fe2bef46c1a535f6c/azurerm/resource_arm_key_vault_secret.go#L179). 

Is there another preferred way of achieving that? If it's just the reuse of `resourceArmKeyVaultKeyCreate`, I agree that it could use some refactoring.